src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -296,8 +299,17 @@ public MethodSpec createReadMethod(ClassName className,
     for (Property prop : adapters.keySet()) {
       FieldSpec field = FieldSpec.builder(prop.type, prop.name).build();
       fields.put(prop, field);
-
-      readMethod.addStatement("$T $N = null", field.type.isPrimitive() ? field.type.box() : field.type, field);
+      if(field.type.isPrimitive()){
```

Comment:
Please fix up the code format.  There seem to be inconsistent whitespace around the if/else blocks.